### DISCUSSION TITLE
Get Diksha Lesson plans

### DISCUSSION THREAD

**Discussion by snehal0904** [Sunbird-inQuiry/Community#16] [General Discussions]
Hello Team,

In the Shiksha platform, we are building a use case where teachers can get the lesson plans for teaching. For this, we wanted to add lesson plans from the Shiksha backend and also get lesson plans created in the Diksha.

From the Diksha document, lesson plans are getting saved in the content with a lesson plan category. We tried to get it using API but there is no data for the lesson plan.
Do we have Lesson plan data on Diksha? Can someone please share the API end point to get this?

Thank you!!

@Krishnaj20 @pallakartheekreddy @coolbung @ananya-jain1 @RinkleDang

**Comment by pramodkvarma** [comment]
As I understand lesson plans are nothing but a type of collection/content. So, same content/collection APIs should work. @vrayulu

**Reply by vrayulu** [reply]
Yes... LessonPlan will be a category. existing content, collection and search APIs should be used.

@snehal0904 - is the category (LessonPlan) created in the environment you are using? and is the content/collection published?

**Reply by snehal0904** [reply]
@vrayulu I am trying to get Lesson plans content from the environment https://vdn.diksha.gov.in. When I call content API on this environment, Now I am getting data with contentType as LessonPlanResource. Is it correct?

**Reply by vrayulu** [reply]
What is the "primaryCategory" value for that content? It should be "LessonPlan".
The contentType attribute is retained for backward compatibility of older apps. You can ignore it and it is correct as long as the "primaryCategory" attribute has the correct value.

**Reply by vrayulu** [reply]
@snehal0904 - if possible, please join the "Knowlg Friday" session scheduled for 3PM today. The session is about object types and primary categories, it may help resolve some of your queries.

Details of the session are available here: https://github.com/orgs/Sunbird-Knowlg/discussions/20
